sections/errorhandling/correlationid.md

```diff
@@ -0,0 +1,35 @@
+# Correlation ID: help your logs tell you a story and give your error a context
+
+### One Paragraph Explainer
+
+Correlation ID is one of the best problem-solving patterns. It lets you link log records, even if they belong to different services. If your system consumes other services and is itself a producer service, adding a correlaction ID is a must. By this pattern, your transaction logs can become into a story that tells itself by filtering your logs with a specific correlation ID, instead of try linking the cross-services transaction logs to each other by yourself. It can save your day when a process including 20 different microservices throws an exception in one of them, and you have no idea where did the problem start across the flow.
```

Comment:
Great writing.

I suggest that the 'microservice' thing should not be the primary point rather secondary (less people do Microservice than we thing), where the main point is correlating entries from single Node.js process across async functions. For example, how to tell this sentence from the api/controller layer belongs to the same request as that entry from the DAL